src/sentry/models/dashboard.py

```diff
@@ -55,6 +69,24 @@ class Meta:
 
     __repr__ = sane_repr("organization", "title")
 
+    @property
+    def favourited_by(self):
+        user_ids = DashboardFavouriteUser.objects.filter(dashboard=self).values_list(
+            "user_id", flat=True
+        )
+        return user_ids
+
+    @favourited_by.setter
+    def favourited_by(self, user_ids):
+        from django.db import router, transaction
+
+        with transaction.atomic(using=router.db_for_write(DashboardFavouriteUser)):
+            DashboardFavouriteUser.objects.filter(dashboard=self).exclude(
+                user_id__in=user_ids
+            ).delete()
+            for user_id in user_ids:
+                DashboardFavouriteUser.objects.get_or_create(dashboard=self, user_id=user_id)
```

Comment:
This is going to make a query for each of the users that have favourited a particular dashboard. Can we find the users that need a relationship created and use `.bulk_create` to do it in one query? i.e. ignore the ones that don't need to be touched?

```py
existing_user_ids = DashboardFavoriteUser.objects.filter(dashboard=self).values_list("user_id", flat=True)
with transaction...:
  DashboardFavoriteUser.objects.filter(dashboard=self).exclude(user_id__in=user_ids)
  
  newly_favourited = [DashboardFavoriteUser(dashboard=self, user_id=user_id) for user_id in set(user_ids) - set(existing_user_ids)]
  DashboardFavoriteUser.objects.bulk_create(newly_favourited)
```
or something like that?

Reply:
Also just a thought, it seems like we're always going to have to pass along all of the `user_ids` when setting this field. Dan mentioned earlier it might make more sense to have a `add_favorite_user` and `remove_favorite_user` APIs. e.g. when you make an endpoint to support `POST /dashboards/favourite` or something like that, you'll have access to the user making the request. If you need to always supply the list of users you'll have to incur and extra query each time.

Is there a use case for needing this kind of bulk update?

Reply:
So if we're using `.bulk_create,` we would still need to `.delete()` the old relationships. I added this so we don't make a query for the users that don't need to be touched:

```python
user_ids_to_delete = list(
                DashboardFavoriteUser.objects.filter(dashboard=self)
                .exclude(user_id__in=user_ids)
                .values_list("user_id", flat=True)
            )
            DashboardFavoriteUser.objects.filter(
                dashboard=self, user_id__in=user_ids_to_delete
            ).delete()
            for user_id in user_ids:
                if user_id not in user_ids_to_delete:
                    DashboardFavoriteUser.objects.get_or_create(dashboard=self, user_id=user_id)
```

Reply:
> So if we're using .bulk_create, we would still need to .delete() the old relationships

I forgot to include the `.delete()` but my intention was that we should keep that too. I'm trying to avoid looping multiple times and creating objects with `.get_or_create`, `.bulk_create` takes a list of items and makes them in one database update

> I think it makes sense to have the settr always bulk update since it's more complex to do that if we needed to

Is there a scenario where we need to do bulk update? I'd rather us not have code that goes unused, but if there is a scenario where we want to do that/have to add this then it makes sense. I'm okay with committing this, but if we can keep in mind whether or not we need it later that would be good

Reply:
Makes sense. I updated the `settr` to use `.bulk_create`, and I'll commit the code for now. Will keep the usage of the `settr` function in mind and get back to it to see if we still need it. Thanks!